iaso/api/profiles.py

```diff
@@ -407,13 +407,23 @@ def update_user_profile(self, request, pk):
         return profile
 
     @staticmethod
-    def update_permissions(user, request):
+    def update_permissions(self, user, request):
         user.user_permissions.clear()
+        current_account = user.iaso_profile.account
+
         for permission_codename in request.data.get("user_permissions", []):
-            CustomPermissionSupport.assert_right_to_assign(request.user, permission_codename)
-            user.user_permissions.add(get_object_or_404(Permission, codename=permission_codename))
+            if permission_codename in self.module_permissions(current_account):
+                CustomPermissionSupport.assert_right_to_assign(request.user, permission_codename)
+                user.user_permissions.add(get_object_or_404(Permission, codename=permission_codename))
         user.save()
 
+    @staticmethod
+    def module_permissions(current_account):
+        # Get all modules linked to the current account
+        modules = list(Module.objects.filter(account_modules=current_account).values_list("id", flat=True))
```

Comment:
Wondering why you needed a list() here, this will most probably add one sql request while using directly the queryset in the following line would have probably given a nested select (so, just  one request). 

These are still probably very small performance impacts, so this does not really need to change.